Release 1.1.0 (#3131)
pokey/cursorless-vscode: packages/cursorless-org-docs/src/docs/user/release-notes/1.1.0.md

### New language support

- **R language support** is now enabled ([#2721](https://github.com/cursorless-dev/cursorless/pull/2721)). Statistics enthusiasts, rejoice!
- **Java properties files** (`.properties`) now have Cursorless support ([#3077](https://github.com/cursorless-dev/cursorless/pull/3077))
- **Talon list files** (`.talon-list`) are now supported ([#3080](https://github.com/cursorless-dev/cursorless/pull/3080))

### New scopes and features

- **Full line scope**: The new `"full line"` scope includes leading and trailing whitespace, unlike the existing `"line"` scope which excludes surrounding whitespace. Try `"change full line"` when you want to grab the whole enchilada. ([#3095](https://github.com/cursorless-dev/cursorless/pull/3095))
- **Argument list scope**: Target the entire argument list of a function with `"arg list"` ([#2907](https://github.com/cursorless-dev/cursorless/pull/2907))
- **Java record statements**: Java records now have proper scope support ([#3089](https://github.com/cursorless-dev/cursorless/pull/3089))

> - **Allow duplicate targets for bring**: You can now bring the same target multiple times ([#3083](https://github.com/cursorless-dev/cursorless/pull/3083))

Author: prob too technical to call out?

Reviewer: Agreed

> - **Move where destination contains source**: More flexible move action behavior ([#3056](https://github.com/cursorless-dev/cursorless/pull/3056))

Author: ```suggestion
- **`"move"` where destination contains source**: This now works ([#3056](https://github.com/cursorless-dev/cursorless/pull/3056))
```

> - **From action enabled by default** ([#2939](https://github.com/cursorless-dev/cursorless/pull/2939))

Author: ```suggestion
- **`"from"` action enabled by default** ([#2939](https://github.com/cursorless-dev/cursorless/pull/2939))
```

> - **Next/previous support for custom commands** ([#2969](https://github.com/cursorless-dev/cursorless/pull/2969))

Reviewer: Probably not. Removed now.

### Documentation and visualization improvements

The docs got some serious love this cycle:

> - **Scope test visualization**: The [language pages](https://www.cursorless.org/docs/user/languages) now show visualizations of scope tests, making it easier to understand exactly what each scope targets ([#3016](https://github.com/cursorless-dev/cursorless/pull/3016))

Author: ```suggestion
- **Language-specific scope visualizations in docs**: The [language pages](https://www.cursorless.org/docs/user/languages) now show visualizations of what each scope looks like in each language, making it easier to understand exactly what each scope targets ([#3016](https://github.com/cursorless-dev/cursorless/pull/3016))
```

- **Missing scopes contributor page**: A new page helps contributors identify which scopes are missing for each language ([#2978](https://github.com/cursorless-dev/cursorless/pull/2978))

> - **Containment icons in scope visualizer**: The scope visualizer now shows containment relationships with icons ([#3085](https://github.com/cursorless-dev/cursorless/pull/3085))

Author: screenshot?

Reviewer: Added

### Interior scope improvements

We made significant improvements to interior scopes (the part of a scope between its delimiters):

> - Added interior scopes to JavaScript, Java, C#, C, and C++ ([#2987](https://github.com/cursorless-dev/cursorless/pull/2987), [#2988](https://github.com/cursorless-dev/cursorless/pull/2988), [#2991](https://github.com/cursorless-dev/cursorless/pull/2991), [#2992](https://github.com/cursorless-dev/cursorless/pull/2992))

Reviewer: In most cases, but I think we can leave this.

- Head/tail modifiers are now bounded by all interior types, not just pairs ([#3070](https://github.com/cursorless-dev/cursorless/pull/3070))
- Changed interior scope handler priority to favor surrounding pair interiors ([#3068](https://github.com/cursorless-dev/cursorless/pull/3068))

### Performance improvements

> - Added scope handler cache for improved performance with multiple cursors ([#3101](https://github.com/cursorless-dev/cursorless/pull/3101))

Reviewer: Added

### Bug fixes

- Fixed bug where move source was incorrectly dropped ([#3061](https://github.com/cursorless-dev/cursorless/pull/3061))
- Fixed R arguments issue with incidental multiple matches ([#3064](https://github.com/cursorless-dev/cursorless/pull/3064))
- Better handling of quick pick with whitespace ([#3045](https://github.com/cursorless-dev/cursorless/pull/3045))
- Don't yield empty iteration scopes ([#2968](https://github.com/cursorless-dev/cursorless/pull/2968))

> ### Support for VS Code forks

Author: these are all just for contributors right?

Reviewer: Yes. I have removed this section.

Cursorless now supports VS Code forks like VSCodium:

- Added sandbox setup command for VSCodium ([#2989](https://github.com/cursorless-dev/cursorless/pull/2989))
- Updated install scripts to support VS Code forks ([#3096](https://github.com/cursorless-dev/cursorless/pull/3096))

> ## 📈 What's next?

Reviewer: Done

- **Cursorless everywhere**: We're continuing work on bringing Cursorless to more editors and platforms.

> - **Embedded languages**: Support for embedded languages like JavaScript within HTML, markdown code blocks, vue.js, styled components, etc. remains a priority ([#409](https://github.com/cursorless-dev/cursorless/issues/409))

Reviewer: Probably not. Removed.
